src/react/anchors-as-client-side-navigation.tsx

```diff
@@ -0,0 +1,67 @@
+import { useNavigate } from "@remix-run/react";
+import { createContext, ReactNode, useContext, useEffect, useRef } from "react";
+
+export interface TreatAnchorsAsClientSideNavigationProps {
+  children: ReactNode;
+}
+
+const HasParentContext = createContext(false);
+
+export function TreatAnchorsAsClientSideNavigation({
+  children,
+}: TreatAnchorsAsClientSideNavigationProps) {
+  const navigate = useNavigate();
+  const ref = useRef<HTMLDivElement>(null);
+  const hasParent = useContext(HasParentContext);
+
+  useEffect(() => {
+    if (hasParent) {
+      console.error(
+        "TreatAnchorsAsClientSideNavigation can't be nested inside another TreatAnchorsAsClientSideNavigation. Ignoring this instance."
+      );
+      return;
+    }
+
+    const controller = new AbortController();
+    ref.current?.addEventListener(
+      "click",
+      (event) => {
+        const target = (event.target as Partial<HTMLElement>).closest?.("a");
```

Comment:
Browser support for `closest` is pretty good but not perfect. Is it ok to use or should I replace with a manual parent search?

https://caniuse.com/element-closest

Reply:
Compared it to [ESM support](https://caniuse.com/es6-module) it has a better support, so it should work on any Remix app.